tests/components/camera/test_push.py

```diff
@@ -0,0 +1,68 @@
+"""The tests for generic camera component."""
+import io
+
+from datetime import timedelta
+from unittest.mock import patch
+
+from homeassistant import core as ha
+from homeassistant.setup import async_setup_component
+from homeassistant.util import dt as dt_util
+from tests.components.auth import async_setup_auth
+
+
+async def test_bad_posting(aioclient_mock, hass, aiohttp_client):
+    """Test that posting to wrong api endpoint fails."""
+    with patch('homeassistant.components.camera.push.PushCamera._blank_image',
+               return_value=io.BytesIO(b'fakeinit')):
+        await async_setup_component(hass, 'camera', {
+            'camera': {
+                'platform': 'push',
+                'name': 'config_test',
+            }})
+
+    client = await async_setup_auth(hass, aiohttp_client)
```

Comment:
What does this do?

Reply:
to be honest I copied the aiohttp_client from `test_api` 

this is supposed to set the environment to make an authenticated post request.